bugsnag-android-core/src/main/java/com/bugsnag/android/Configuration.java

```diff
@@ -0,0 +1,836 @@
+     *
+     * To provide custom delivery functionality, create a class which implements the Delivery
+     * interface. Please note that request bodies must match the structure specified in the
+     * <a href="https://docs.bugsnag.com/api/error-reporting/">Error Reporting</a> and
+     * <a href="https://docs.bugsnag.com/api/sessions/">Sessions API</a> documentation.
+     *
+     * You can use the return type from the deliver functions to control the strategy for
+     * retrying the transmission at a later date.
+     *
+     * If DeliveryStatus.UNDELIVERED is returned, the notifier will automatically cache
+     * the payload and trigger delivery later on. Otherwise, if either DeliveryStatus.DELIVERED
+     * or DeliveryStatus.FAILURE is returned the notifier will removed any cached payload
+     * and no further delivery will be attempted.
+     */
+    public void setDelivery(@NonNull Delivery delivery) {
+        if (delivery != null) {
+            impl.setDelivery(delivery);
+        } else {
+            logNull("delivery");
+        }
+    }
+
+    /**
+     * Set the endpoints to send data to. By default we'll send error reports to
+     * https://notify.bugsnag.com, and sessions to https://sessions.bugsnag.com, but you can
+     * override this if you are using Bugsnag Enterprise to point to your own Bugsnag endpoints.
+     */
+    @Nullable
+    public EndpointConfiguration getEndpoints() {
+        return impl.getEndpoints();
+    }
+
+    /**
+     * Set the endpoints to send data to. By default we'll send error reports to
+     * https://notify.bugsnag.com, and sessions to https://sessions.bugsnag.com, but you can
+     * override this if you are using Bugsnag Enterprise to point to your own Bugsnag endpoints.
+     */
+    public void setEndpoints(@Nullable EndpointConfiguration endpoints) {
+        if (endpoints != null) {
+            impl.setEndpoints(endpoints);
+        } else {
+            impl.setEndpoints(new EndpointConfiguration());
+        }
+    }
+
+    /**
+     * Sets the maximum number of breadcrumbs which will be stored. Once the threshold is reached,
+     * the oldest breadcrumbs will be deleted.
+     *
+     * By default, 25 breadcrumbs are stored: this can be amended up to a maximum of 100.
+     */
+    public int getMaxBreadcrumbs() {
+        return impl.getMaxBreadcrumbs();
+    }
+
+    /**
+     * Sets the maximum number of breadcrumbs which will be stored. Once the threshold is reached,
+     * the oldest breadcrumbs will be deleted.
+     *
+     * By default, 25 breadcrumbs are stored: this can be amended up to a maximum of 100.
+     */
+    public void setMaxBreadcrumbs(int maxBreadcrumbs) {
+        if (maxBreadcrumbs >= MIN_BREADCRUMBS && maxBreadcrumbs <= MAX_BREADCRUMBS) {
+            impl.setMaxBreadcrumbs(maxBreadcrumbs);
+        } else {
+            getLogger().e("Invalid null value supplied to config.maxBreadcrumbs, expected "
```

Comment:
There's an error message format in the spec - see Configuration page.

Reply:
updated to follow message format